### ISSUE TITLE
Feature request: wildcard in patterns

### ISSUE THREAD

**Issue by protz** [CatalaLang/catala#105]
It would be convenient to be able to do:

```
match e₁ with pattern
  -- * : e₂
```

the syntax can also be an underscore, although maybe a star will be more understandable ?

**Comment by denismerigoux** [comment]
The location to change to add this feature is
 
https://github.com/CatalaLang/catala/blob/2cad9823f74ebc2fb13dd90a061e9bb4fbcb4da1/src/catala/surface/desugaring.ml#L394-L400

**Comment by EmileRolley** [comment]
> It would be convenient to be able to do:
> 
> ```
> match e₁ with pattern
>   -- * : e₂
> ```
> 
> the syntax can also be an underscore, although maybe a star will be more understandable ?

It could be the keyword `any` (respectively `n'importe quel` for the french syntax).

**Comment by denismerigoux** [comment]
Actually we should ask lawyers about this keyword too :)

**Comment by EmileRolley** [comment]
To be sure, the wildcard should match all cases or all remaning cases?

For example, with this declarations:

```
declaration enumeration E:
  -- Case1 content integer
  -- Case2
  -- Case3

declaration scope A:
  context x content E
  context y content integer
```

Should this code be considered correct?
```
scope A:
  definition x equals Case1 content 2
  definition y equals match x with pattern
    -- Case1 of i : 42
    -- anything : 43
```
And equivalent to:
```
scope A:
  definition x equals Case1 content 2
  definition y equals match x with pattern
    -- Case1 of i : 42
    -- Case2 : 43
    -- Case3 : 43
```

Or should it only be used alone?

```
scope A:
  definition x equals Case1 content 2
  definition y equals match x with pattern
    -- anything : 42
```

**Comment by denismerigoux** [comment]
The wildcard should match remaining cases, as in nearly all languages with pattern matching.

**Comment by denismerigoux** [comment]
As the  pattern matching in `Scopelang`, `Dcalc` and `LCalc` don't support it, you can either add support for wildcard pattern matching in all of those languages, or you can do a workaround by let-binding the wildcard arm expression just before the match and putting the reference to it in all the branches the wildcard stands for.

**Comment by denismerigoux** [comment]
This latter option should not change the semantics because there are no side-effects when evaluating expression that don't have defaults in Catala, but ideally we should add wildcard pattern matching support all the way down.

**Comment by EmileRolley** [comment]
Closed by #129
